### DISCUSSION TITLE
"All checks have passed" only taking latest commit into account

### DISCUSSION THREAD

**Discussion by sj14** [community/community#27267] [Actions] [answered]
# Description
When you have Github Workflows which only run on specific file changes (`on.push.paths`) and a commit leads to a failed workflow, this failed workflow is ignored (in the PR overview and `Checks` tab) when there are follow-up commits which will not trigger the same (failed) workflow.

# Example

### Overview
![1commit_overview](https://user-images.githubusercontent.com/6493966/181199375-37a72013-5e85-4f8f-b072-23863c3f471b.png)

### Commits
![Bildschirmfoto 2022-07-27 um 10 21 53](https://user-images.githubusercontent.com/6493966/181199569-d1b40762-0708-48e4-ab9b-22ca38895bb0.png)

### Overview
![Ohne Titel](https://user-images.githubusercontent.com/6493966/181200260-0e8cc724-bba3-434b-b2fc-b8301ec6aaa8.png)

### Commits
![Bildschirmfoto 2022-07-27 um 10 23 21](https://user-images.githubusercontent.com/6493966/181199448-1270b7a4-f722-4d9b-9f76-be3268c1c251.png)

# Conclusion and suggestion

Thus, you could easily think all checks were successful and merge a PR but it only reflects the checks from the latest commit in this PR.

I would suggest always showing the latest results from all workflows that were run in a PR in the overview and `Checks` tab.

**Comment by sigprof** [comment, marked as answer]
You probably need to change the problematic workflow to run on `pull_request` — then it would be rerun on any changes to the PR, even when the last commit does not change the specified paths, but some previous ones do.

Also there is a subtle but very important difference between `push` and `pull_request` — workflows running on `push` test exactly the code that was pushed, but workflows running on `pull_request` test the result of attempted merge between the PR target and head branches. So the `push` workflows will never detect that the code in the PR head branch is not actually compatible with some other changes made in the PR target branch — only the `pull_request` workflows can detect that.

**Reply by sj14** [reply]
Unfortunately, it was a conscious decision to use `push` to decrease the time the pipeline takes (many executed jobs were unnecessary when unrelated files changed). Except for the given issue, it seems to work fine so far.

**Reply by sj14** [reply]
Ah, I see. `on` can be used with `pull_request` too. I will test it 👍

**Reply by shaneDubbs** [reply]
This solution does not fit at all for situations where the pull request trigger is limited by paths. Some of our checks take a rather long time and we only want to run them when certain files and their dependencies change not when some different file changes and it so happened to be in a previous commit.

**Reply by farrukh-taqveem-carv** [reply]
@sj14 Did you find a solution.

**Reply by sj14** [reply]
@farrukh-taqveem-carv  I use `on.pull_request.paths` with the trade off @shaneDubbs described, and it only triggers for pull requests but not branches without a correspondign pull requests.
